### DISCUSSION TITLE
Properly passing in dynamic custom props to variants

### DISCUSSION THREAD

**Discussion by cmco** [chakra-ui/chakra-ui#5644] [Q&A]
Given a variant of a component (e.g. Button), is there a way to properly dynamically pass in a prop?

I.e.
Having a component such as `<Button variant="customVariant" textColor={textColor} />` would have the corresponding style in a custom theme file:

```
variants: {
    customVariant: ({ textColor }) => ({
       color: textColor || 'black',
    )}
}      
```

I mentioned "properly" since this method already works but throws a warning in the console saying:

``React does not recognize the `textColor` prop on a DOM element``

Any insight would be appreciated. Thanks!

**Comment by mawais786** [comment]
@cmco have you fixed that issue?
